python/ray/data/aggregate.py

```diff
@@ -312,3 +343,79 @@ def _to_on_fn(on: Optional[KeyFn]):
         return lambda r: r[on]
     else:
         return on
+
+
+@PublicAPI
+class Quantile(_AggregateOnKeyBase):
+    """Defines Quantile aggregation."""
+
+    def __init__(self, on: Optional[KeyFn] = None, percent: float = 0.5, ignore_nulls: bool = True,
+                 alias_name: Optional[KeyFn] = None):
+        self._set_key_fn(on)
+        self._percent = percent
+
+        def merge(a: List[int], b: List[int]):
+            if isinstance(a, List) and isinstance(b, List):
+                a.extend(b)
+                return a
+            if isinstance(a, List) and (not isinstance(b, List)):
+                if b is not None and b != '':
+                    a.append(b)
+                return a
+            if isinstance(b, List) and (not isinstance(a, List)):
+                if a is not None and a != '':
+                    b.append(a)
+                return b
+
+            ls = []
+            if a is not None and a != '':
+                ls.append(a)
+            if b is not None and b != '':
+                ls.append(b)
+            return ls
+
+        null_merge = _null_wrap_merge(ignore_nulls, merge)
+
+        def block_row_ls(block: Block[T]) -> AggType:
+            block_acc = BlockAccessor.for_block(block)
+            ls = []
+            for row in block_acc.iter_rows():
+                ls.append(row.get(on))
+            return ls
+
+        if alias_name:
+            rs_name = alias_name
+        else:
+            rs_name = (f"Quantile({str(on)})")
+
+        import math
+        def percentile(N, key=lambda x: x):
+            if not N:
+                return None
+            N = sorted(N)
+            k = (len(N) - 1) * self._percent
+            f = math.floor(k)
+            c = math.ceil(k)
+            if f == c:
+                return key(N[int(k)])
+            d0 = key(N[int(f)]) * (c - k)
+            d1 = key(N[int(c)]) * (k - f)
+            return round(d0 + d1, 5)
+
+        super().__init__(
+            init=_null_wrap_init(lambda k: [0]),
+            merge=null_merge,
+            accumulate_block=_null_wrap_accumulate_block(
+                ignore_nulls,
+                block_row_ls,
+                null_merge,
+            ),
+            finalize=_null_wrap_finalize(percentile),
+            name=(rs_name),
+        )
+
```

Comment:
Some lint issues (run scripts/format.sh to fix): https://buildkite.com/ray-project/oss-ci-build-pr/builds/18407#01877969-e28d-42df-b046-0e7a516a22a5

Reply:
Sure, I will execute this sh immediately. Thank you very much